dashi/poolqc/pooling_qc_sample.py

```diff
@@ -216,35 +295,88 @@ def update_sampleindices(run):
+            go.Bar(
+                x=list(d['library']),
+                y=list(d['Per Cent Difference']),
+                name=inx,
+                marker={'color': '#20639B'},
+            )
+        )
+    return {
+        'data': data,
+        'layout': {
+            'title': 'Per Cent Difference of Index Clusters',
+            'xaxis': {'title': 'Sample', 'automargin': True},
+            'yaxis': {'title': 'Per Cent', 'range': [-100, 100]},
+            'showlegend': False,
+        }
+    }
+
+
 @app.callback(
     [dep.Output('SampleIndices', 'figure'),
+     dep.Output('Per Cent Difference', 'figure'),
      dep.Output('Summary Table', 'columns'),
      dep.Output('Summary Table', 'data'),
-     dep.Output('Summary Table', 'style_data_conditional'),
      dep.Output('download-link', 'href'),
-     dep.Output('download-link', 'download')],
+     dep.Output('download-link', 'download'),
+     dep.Output('object_threshold', 'children'),
+     dep.Output('object_passed_samples', 'children'),
+     dep.Output('object_option3', 'children')],
     [dep.Input('select_a_run', 'value'),
-     dep.Input('lane_select', 'value')])
-def update_graphs(run_alias, lane_alias):
+     dep.Input('lane_select', 'value'),
+     dep.Input('index_threshold', 'value'),
+     dep.Input('pass/fail', 'value'),
+     dep.Input('sample_type', 'value')])
+def update_graphs(run_alias, lane_alias, threshold, PassOrFail, sample_type):
+    index_threshold = int(threshold)
+
     run = df[(df['Run'] == run_alias) & (df['LaneNumber'] == lane_alias)]
-    run = run[~run['library'].isna()].drop_duplicates('library')
+    run = run[~run['Run'].isna()].drop_duplicates('library')
+    run = run[~run['library'].isna()]
+
+    num_libraries = len(run['library'])
+    samples_passing_clusters = '%s/%s' % (sum(i > index_threshold for i in run['Clusters']), num_libraries)
+
+    pass_or_fail = []
+    for row in run['Clusters']:
+        if row >= index_threshold:
+            pass_or_fail.append('Pass')
+        if row < index_threshold:
+            pass_or_fail.append('Fail')
+    run['Pass/Fail'] = pass_or_fail
+    run = run[run['Pass/Fail'].isin(PassOrFail)]
+
+    run['Sample Type'] = run['library'].str[-2:]
+    run = run[run['Sample Type'].isin(sample_type)]
 
     downloadtimedate = datetime.today().strftime('%Y-%m-%d')
     download = 'PoolQC_%s_%s_%s.csv' % (downloadtimedate, run_alias, lane_alias)
 
-    columns, data, style_data_conditional, csv = Summary_table(run)
+    columns, data, csv = Summary_table(run)
 
-    return update_sampleindices(run), columns, data, style_data_conditional, csv, download
+    return update_sampleindices(run, index_threshold), percent_difference(run,
+                                                                          index_threshold), columns, data, csv, download, (
+                   'Threshold: ' + str(index_threshold)), (
+                   'Passed Samples: ' + str(samples_passing_clusters)), 'Option3',
```

Comment:
What is "Option3" for?

Reply:
I was going to use it as a place to display the types of libraries selected, but it ends up being redundant since the filters are pretty self-explanatory. Have removed it